conda/environments/all_cuda-118.yaml

```diff
@@ -18,6 +18,7 @@ dependencies:
 - pydata-sphinx-theme
 - pytest<8
 - python-build>=1.2.0
+- python>=3.8,<3.12
```

Comment:
Since this is to control the version tightly, should we fix it to a specific version here?

Reply:
Yeah, I think you're right! Just pushed https://github.com/rapidsai/legate-boost/pull/140/commits/bf45938a2f48315c42071b788c1764da953bc80d changing this pin to `python=3.11`.

I'd originally introduced a range here because that's what `cudf` does ([code link](https://github.com/rapidsai/cudf/blob/a3dc14fcea938729c7c9468bd6a64331395b2f78/conda/environments/all_cuda-118_arch-x86_64.yaml#L78)).

But thinking about it some more... this file is intended to only be used interactively (not in CI), and isn't intended to be merged with any other environment-creation files. Choosing a single, supported Python version should make conda solves faster.

And anyone who wanted to work on a specific, different version of Python would have had to change this constraint anyway (or generate a different environment file using `rapids-dependency-file-generator` and passing the `py` matrix selector).